app/services/post_status_service.rb

```diff
@@ -52,6 +52,7 @@ def preprocess_attributes!
     @text         = @options.delete(:spoiler_text) if @text.blank? && @options[:spoiler_text].present?
     @visibility   = @options[:visibility] || @account.user&.setting_default_privacy
     @visibility   = :unlisted if @visibility&.to_sym == :public && @account.silenced?
+    @visibility   = :limited if @visibility&.to_sym != :direct && @in_reply_to&.limited_visibility?
```

Comment:
Currently, replying in unlisted/public to a followers-only toot is possible and can be useful, even if the use cases for it are rather rare.

This changes disallows replying in unlisted/public to a limited toot. I understand the reasoning, but it's new behavior, and might be especially confusing since it is done silently. Maybe erroring out would be better.

Reply:
Replying private to limited would be a privacy downgrade (more people see the reply than the original post) which is unintuitive and as I figured, undesirable.